agent/engine/containermetadata/acquire_metadata.go

```diff
@@ -0,0 +1,181 @@
+// Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package containermetadata
+
+import (
+	"github.com/aws/amazon-ecs-agent/agent/api"
+	"github.com/aws/amazon-ecs-agent/agent/config"
+
+	"github.com/cihub/seelog"
+	docker "github.com/fsouza/go-dockerclient"
+)
+
+// acquireNetworkMetadata parses the docker.NetworkSettings struct and
+// packages the desired metadata for JSON marshaling
+func acquireNetworkMetadata(settings *docker.NetworkSettings) NetworkMetadata {
+	if settings == nil {
+		seelog.Errorf("Failed to acquire network metadata: Network metadata not available or does not exist")
+		return NetworkMetadata{}
+	}
+
+	// This metadata is available in two different places in NetworkSettings
+	ipv4Address := settings.IPAddress
+	ipv4Gateway := settings.Gateway
+	ipv6Address := settings.GlobalIPv6Address
+	ipv6Gateway := settings.IPv6Gateway
+
+	// Network mode is not available for Docker API versions 1.17-1.20
+	networkModeFromContainer := ""
+	if len(settings.Networks) == 1 {
+		for modeFromSettings, containerNetwork := range settings.Networks {
+			networkModeFromContainer = modeFromSettings
+			ipv4Address = containerNetwork.IPAddress
+			ipv4Gateway = containerNetwork.Gateway
+			ipv6Address = containerNetwork.GlobalIPv6Address
```

Comment:
After investigating this issue further, I have found a valid use case where ipv6 addresses will appear in the container's metadata. 
Running docker with `--ipv6 --fixed-cidr-v6="2001:db8:1::/64"`
https://docs.docker.com/engine/userguide/networking/default_network/ipv6/#how-ipv6-works-on-docker
Then doing a `docker inspect` for a container on `bridge` mode gives 
```
"SecondaryIPv6Addresses": null,
            "EndpointID": "c0fc99640eac868b1d415dddcb60d575a1aa93e61e71eb0183334c13443875d5",
            "Gateway": "172.17.0.1",
            "GlobalIPv6Address": "2001:db8:1::242:ac11:2",
            "GlobalIPv6PrefixLen": 64,
            "IPAddress": "172.17.0.2",
            "IPPrefixLen": 16,
            "IPv6Gateway": "2001:db8:1::1",
            "MacAddress": "02:42:ac:11:00:02",
            "Networks": {
                "bridge": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": null,
                    "NetworkID": "018648e9c2314ef09df71da4e1074cfb98f8a164823b95add453d14b7c4dfcf3",
                    "EndpointID": "c0fc99640eac868b1d415dddcb60d575a1aa93e61e71eb0183334c13443875d5",
                    "Gateway": "172.17.0.1",
                    "IPAddress": "172.17.0.2",
                    "IPPrefixLen": 16,
                    "IPv6Gateway": "2001:db8:1::1",
                    "GlobalIPv6Address": "2001:db8:1::242:ac11:2",
                    "GlobalIPv6PrefixLen": 64,
                    "MacAddress": "02:42:ac:11:00:02"
                }
            }

```